tests/unit/core/test-class-wc-payments-customer-service-api.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Class WC_Payments_Customer_Service_Test
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use PHPUnit\Framework\MockObject\MockObject;
+use WCPay\Database_Cache;
+use WCPay\Exceptions\API_Exception;
+
+/**
+ * WC_Payments_Customer_Service unit tests.
+ */
+class WC_Payments_Customer_Service_API_Test extends WCPAY_UnitTestCase {
+
+	const CUSTOMER_LIVE_META_KEY = '_wcpay_customer_id_live';
+	const CUSTOMER_TEST_META_KEY = '_wcpay_customer_id_test';
+
+	/**
+	 * System under test.
+	 *
+	 * @var WC_Payments_Customer_Service_API
+	 */
+	private $customer_service_api;
+
+	/**
+	 *
+	 * @var WC_Payments_Customer_Service
+	 */
+	private $customer_service;
+
+	/**
+	 * Mock WC_Payments_Http.
+	 *
+	 * @var WC_Payments_Http|MockObject
+	 */
+	private $mock_http_client;
+
+
+	/**
+	 * Pre-test setup
+	 */
+	public function set_up() {
+		$this->mock_http_client = $this
+			->getMockBuilder( 'WC_Payments_Http' )
+			->disableOriginalConstructor()
+			->setMethods( [ 'get_blog_id', 'is_connected', 'remote_request' ] )
+			->getMock();
+		add_filter(
+			'wc_payments_http',
+			function() {
+				return $this->mock_http_client;
+			}
+		);
+		$this->customer_service     = new WC_Payments_Customer_Service( WC_Payments::create_api_client(), WC_Payments::get_account_service(), WC_Payments::get_database_cache() );
```

Comment:
I didn't realize `create_api_client` is public, and available to use this way for mocks, without replacing the main API client. I'll keep this approach in mind for future work. Thank you!